### PR TITLE
Fixed almost all MSVC 2019 Warnings

### PR DESCRIPTION
**Pull request by FlorianReimold**
Fixed almost all MSVC 2019 Warnings:
- Rec
- Sys
- Mon
- Doc Core
- Samples

### CODE REVIEW

**Comment by FlorianReimold** [comment]
Last thing to fix is this one:

https://github.com/continental/ecal/blob/cb9956d52449e4a04c728652b6da08df53ab7939/ecal/core/src/service/ecal_tcpclient.cpp#L299

The function seems very sketchy and doesn't appear to do what it is expected to do. The timeout parameter is unused.
@rex-schilasky / @brakmic-aleksandar : How shall I fix this? What is the function supposed to do?

**Comment by rex-schilasky** [comment]
> Last thing to fix is this one:
> 
> https://github.com/continental/ecal/blob/cb9956d52449e4a04c728652b6da08df53ab7939/ecal/core/src/service/ecal_tcpclient.cpp#L299
> 
> The function seems very sketchy and doesn't appear to do what it is expected to do. The timeout parameter is unused. @rex-schilasky / @brakmic-aleksandar : How shall I fix this? What is the function supposed to do?

This function is not implemented fully. It's doing (as the name says) a receive for a server response asynchronously (returns immediately). The timeout parameter is not supported yet, see API doc

https://github.com/continental/ecal/blob/87fefd756b0542da3c40a1d26799d5f8a7d3906c/ecal/core/include/ecal/ecal_client.h#L134

**Comment by FlorianReimold** [comment]
eCAL now is _basically_ warning free in Visual Studio and GCC11

grafik

**Comment by Kerstin-Keller** [comment]
I guess we could check the python version and call only if python <= 3.6? (via `#if`)?

File: lib/ecal_utils/src/win_cp_changer.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ namespace EcalUtils
   WinCpChanger::WinCpChanger(const int target_cp)
   {
     initial_cp = GetConsoleOutputCP();
-    SetConsoleOutputCP(CP_UTF8);
+    SetConsoleOutputCP(target_cp);
   }
```
Comment (Kerstin-Keller) [inline lib/ecal_utils/src/win_cp_changer.cpp:36]: This is actually a functional change? If you want to fix the warnings only I would do `const int /*target_cp*/` and do the behavioral change in a different PR? What does thins change imply (e.g. what is passed as parameter in it's usage).?

Comment (FlorianReimold) [reply lib/ecal_utils/src/win_cp_changer.cpp:36]: It is a functional change, but it was always supposed to function that way. I just didn't realize the function completely ignores the parameter because I only use it to switch to UTF-8 anyways.

**Comment by FlorianReimold** [comment]
I fixed the Python Warnings in a separete Branch / PR: #606 

The reason is that I want to Squash-merge this commit but cherry-pick the Python changes into 5.9 and 5.8. That way we are safe to also compile 5.9 against python 3.11 in the future.

**Review by rex-schilasky — APPROVED** [review]
